### DISCUSSION TITLE
Repository is not prepared

### DISCUSSION THREAD

**Discussion by MartkCz** [cycle/orm#495] [Q&A] [answered]
I get this weird error occasionally and randomly. I use spiral framework with roadrunner. ORMInterface is not null in my RepositoryAccessor, but it's in RepositoryInterface. Where's the problem?

![image](https://github.com/user-attachments/assets/9c402965-b43d-4547-a2b5-d4329fbe5af9)

**Comment by roxblnfk** [comment, marked as answer]
Hi. Do you use `warmup` option in ORM config? Try to switch this one.

**Reply by MartkCz** [reply]
No, I used the default application skeleton. It seems to work.

```php
<?php declare(strict_types=1);

// @see https://spiral.dev/docs/basics-orm/current/en#orm

return [
	'schema' => [
		'cache' => true,
	],
	/**
	 * Prepare all internal ORM services (mappers, repositories, typecasters...)
	 */
	'warmup' => env('APP_ENV') === 'prod',
];

```

Thanks for the help

**Reply by roxblnfk** [reply]
Could you clarify which value it operates with? `false`?

**Reply by MartkCz** [reply]
Production: true, dev: false. I didn't have "cycle.php" file in the project before. Now all environments work properly :)

**Reply by roxblnfk** [reply]
Note that `RepositoryProvider` returns a repository based on the entity's **role**.
In ORM, this service is used as follows:
```php
    public function getRepository(string|object $entity): RepositoryInterface
    {
        return $this->repositoryProvider->getRepository(
            $this->resolveRole($entity)
        );
    }
```

If you want to use `RepositoryProvider` directly, you should resolve the entity role using `$orm->resolveRole($entity)`

**Reply by MartkCz** [reply]
Nice note! I am used to it from doctrine. I'll change the code, thanks!

**Reply by roxblnfk** [reply]
Yeah, it's preferred to use ORM interface instead of its services because the ORM interface resolves roles automatically.
